src/lib.rs

```diff
@@ -548,6 +549,62 @@ impl<T> Grid<T> {
         }
     }
 
+    /// Access a certain row on the grid.
+    /// Returns `None` if a row beyond the grid bounds is tried to be accessed,
+    /// or if the grid is column-major.
+    #[must_use]
+    pub fn get_row(&self, row: impl TryInto<usize>) -> Option<&[T]> {
+        let row_usize = row.try_into().ok()?;
+        if row_usize < self.rows && self.order == Order::RowMajor {
+            let start = row_usize * self.cols;
+            Some(self.data[start..(start + self.cols)].as_ref())
+        } else {
+            None
```

Comment:
I had a look at the changes. Thanks for the formatting. I better setup my auto-formatter better.

I wonder if the implementaiton helps or is more confusing. I don't think it is a good idea to expose a function called `get_row` which only works in `RowMajor` mode.

Why not use the already existing `iter_row` funciton instead and let the caller decide if collecting the result into a `Vec` is desired?